src/App.tsx

```diff
@@ -0,0 +1,345 @@
+import UnAuthenticatedModal from "./components/UnAuthenticatedModal";
+import UnAuthorizedModal from "./components/UnAuthorizedModal";
+
+
+
+const App = (props: RouteComponentProps & DispatchProps) => {
+
+  const {
+    fetchLizardBootstrap,
+    fetchOrganisations,
+    getDatasets,
+    updateTaskStatus,
+    addNotification,
+  } = props;
+
+  const userAuthenticated = useSelector(getUserAuthenticated);
+  const isBusyFetchingBootstrap = useSelector(getIsFetchingBootstrap);
+  const shouldFetchBootstrap =  useSelector(getShouldFetchBootstrap);
+  const shouldFetchOrganisations = useSelector(getShouldFetchOrganisations);
+  const selectedOrganisation = useSelector(getSelectedOrganisation);
+  const showOrganisationPicker = userAuthenticated && !shouldFetchOrganisations && selectedOrganisation;
+  const filesInProcess = useSelector( getFilesInProcess);
+  const firstFileInTheQueue = filesInProcess && filesInProcess[0];
+  const currentRelativeUrl = props.location.pathname;
+  
+  const [showOrganisationSwitcher, setShowOrganisationSwitcher] = useState(false);
+  const [showUploadQueue, setShowUploadQueue] = useState(false);
+  const [showUnauthenticatedRedirectModal, setShowUnauthenticatedRedirectModal] = useState(false);
+  const [showUnAuthorizedRedirectModal, setShowUnAuthorizedRedirectModal] = useState(false);
+
+  
+
+  // fetch if user is authenticated
+  useEffect(() => {
+    if (shouldFetchBootstrap) {
+      fetchLizardBootstrap();
+    }
+  }, [fetchLizardBootstrap, shouldFetchBootstrap]);
+  useEffect(() => {
+    if (userAuthenticated && shouldFetchOrganisations) {
+      fetchOrganisations();
+      getDatasets();
+    }
+  }, [userAuthenticated, shouldFetchOrganisations, fetchOrganisations, getDatasets]);
+
+  useEffect(() => {
+    if (firstFileInTheQueue && firstFileInTheQueue.uuid) {
+      setTimeout(() => {
+        fetchTaskInstance(firstFileInTheQueue.uuid)
+          .then(response => {
+            updateTaskStatus(firstFileInTheQueue.uuid, response.status);
+          })
+          .catch(e => console.error(e))
+      }, 5000);
+    }
+  }, [firstFileInTheQueue, updateTaskStatus]);
+
+  const handleWindowClose = useCallback(event => {
+    event.preventDefault();
+    
+    if (filesInProcess && filesInProcess.length > 0) {
+      return event.returnValue = "";
+    } else {
+      return null;
+    };
+  }, [filesInProcess]);
+
+  useEffect(() => {
+    window.addEventListener('beforeunload', handleWindowClose);
+
+    // todo, should other logic concerning a warning on close also be handled here? Or in the forms?
+
+    // removing event listener not needed since the logic is in the eventlistener itself?
+    // return () => {
+    //   window.removeEventListener('beforeunload', handleWindowClose);
+    // };
+  }, [handleWindowClose, filesInProcess]);
+
+  const updateOnlineStatus = useCallback(event => {
+    addNotification(`Your internet connection seems to be ${event.type}`, 2000);
+  },[addNotification])
```

Comment:
Can it be just a normal function instead of using useCallback? What is the benefit of using useCallback here?